pkg/metrics/metrics_context.go

```diff
@@ -0,0 +1,26 @@
+package metrics
+
+import (
+	"context"
+
+	stats "github.com/hellofresh/stats-go"
+)
+
+type statsKeyType int
+
+const statsKey statsKeyType = iota
+
+// NewContext returns a context that has a stats Client
+func NewContext(ctx context.Context, client stats.Client) context.Context {
+	return context.WithValue(ctx, statsKey, client)
+}
+
+// WithContext returns a stats Client with as much context as possible
+func WithContext(ctx context.Context) stats.Client {
+	ctxStats, ok := ctx.Value(statsKey).(stats.Client)
+	if !ok {
+		ctxStats, _ := stats.NewClient("noop://", "")
```

Comment:
Can you bump `stats-go` to the latest version here or should we do this in separate PR?

Reply:
Lets do it in a separate PR